src/dealDashboard/dealDiscussion/discussionThread/discussionThread.ts

```diff
@@ -124,13 +124,25 @@ export class DiscussionThread {
     }, {});
   }
 
+  private streamIDs = new Set();
+
   private async updateCommentsThreadUponMessageArrival(comment: Types.Message): Promise<void> {
     // If a new comment is added to the thread, it is added at the end of the comments array.
-    this.threadDictionary = this.arrayToDictionary(this.threadComments);
+    if (this.streamIDs.has(comment.id)) {
+      return;
+    } else {
+      if (this.streamIDs.size > 10) {
+        const firstEntered = this.streamIDs.values().next();
+        this.streamIDs.delete(firstEntered.value);
 
-    const newComment: IComment = {...comment.data};
+      }
+      this.streamIDs.add(comment.id);
+    }
 
-    if (!this.threadDictionary[newComment._id]) {
+    const newComment: IComment = {...comment.data};
```

Comment:
What's the goal of the spread?

If want a deepCopy, then we might want to add `Utils.cloneDeep() {  json.parse(json.stringify)  }`